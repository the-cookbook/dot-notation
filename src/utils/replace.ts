/**
 * Replace
 * @description replaces string search values
 * @param {string} source
 * @param {string[]} searchValues
 * @param {string} replaceWith
 * @returns {string}
 */
const replace = (source: string, searchValues: string[], replaceWith: string): string =>
  searchValues.reduce((raw, value) => raw.replace(value, replaceWith), source);

export default replace;
