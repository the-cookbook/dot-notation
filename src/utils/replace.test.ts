import replace from './replace';

const source = 'Spicy jalapeno bacon ipsum dolor amet spare ribs ham doner venison ground round strip steak';

describe('utils/replace()', () => {
  it('should match snapshot', () => {
    expect(replace).toMatchSnapshot();
  });

  it('should work with all search value types', () => {
    const search = ['oh-snap', null, undefined];

    expect(replace(source, search, '')).toBe(source);
  });

  it('should return original source value when there is no match', () => {
    const search = ['oh-snap'];

    expect(replace(source, search, '')).toBe(source);
  });

  it('should replace search values with replace value', () => {
    const search = ['ipsum dolor amet'];
    const expects = 'Spicy jalapeno bacon cheese burger spare ribs ham doner venison ground round strip steak';

    expect(replace(source, search, 'cheese burger')).toBe(expects);
  });
});
