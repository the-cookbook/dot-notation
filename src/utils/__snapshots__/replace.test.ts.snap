// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/replace() should match snapshot 1`] = `[Function]`;
